webservices/tasks/download.py

```diff
@@ -76,75 +70,43 @@ def parse_kwargs(resource, qs):
         kwargs = flaskparser.parser.parse(fields)
     return fields, kwargs
 
-def iter_paginator(paginator):
-    last_index, sort_index = (None, None)
-    while True:
-        logger.info(
-            'Fetching page with last_index={last_index}, sort_index={sort_index}'.format(
-                **locals()
-            )
-        )
-        page = paginator.get_page(last_index=last_index, sort_index=sort_index)
-        if not page.results:
-            return
-        for result in page.results:
-            yield result
-        last_indexes = paginator._get_index_values(result)
-        last_index = last_indexes['last_index']
-        if paginator.sort_column:
-            sort_index = last_indexes['last_{}'.format(paginator.sort_column[0].key)]
-        db.session.expunge_all()
-        del page
+def query_to_csv(query, fp):
+    dialect = postgresql.dialect()
+    compiled = query.statement.compile(dialect=dialect)
+    # Get connection from session to hook into replica load balancing
+    conn = db.session.connection().engine.raw_connection()
```

Comment:
Since we're dealing with the database driver directly and bypassing the ORM, we need a raw psycopg2 connection. You can get more details at http://docs.sqlalchemy.org/en/latest/core/connections.html#sqlalchemy.engine.Engine.raw_connection.